rfcs/0168-feature-parameter-names.md

```diff
@@ -0,0 +1,559 @@
+configurations with non-default feature flags. Only configurations that has
+name in package set (e.g `emacs-nox`) SHOULD be built on CI.
+
+## The migration process.
+
+### Orgranization rules
+
+By the very nature of this RFC, once it passes, numerous packages in Nixpkgs
+will become non-compliant. This is unavoidable, otherwise this RFC would not be
+necessary. The following organizational process shall be followed to ensure
+that nixpkgs will eventually become compliant with this RFC:
+
+* In existing packages, maintainers MAY start to expose feature parameters and MAY cease exposing them.
+* Further contributions MAY change their package to be compliant with the RFC
+* Further contributions MUST NOT rename non-compliant feature parameters to another non-compliant name.
+* Further contributions MUST NOT rename compliant feature parameters to non-compliant parameter.
+* Further contributions MAY rename compliant feature parameters to another compliant name
+  for the purposes of consistency with other packages.
+* Further contributions that improve compliance with this RFC MAY involve
+  maintainers but SHOULD NOT be required to do so because the rules SHOULD be
+  clear.
+* New packages MUST be compliant with the RFC (either no feature parameters or compliant with this RFC.)
+
+This ensures that count of non-compliant feature parameters in nixpkgs is
+non-increasing over the time.
+
+### Technical process
+
+Named parameters are part of the programming interface, and renaming them is a
+breaking change. As such, renaming of the parameters is done in following way:
+
+1. Following function is added into `lib` set:
+
+```
+let renamed = { oldName, newName, sunset, oldValue }: newValue:
+   let warning = builtins.concatStringsSep " "
+      [ "Feature flag"
+      , oldName
+      , "is renamed to"
+      , newName
+      , "; old name will no longer be available in nixpkgs="
+      , sunset
+      , "."
+      ]
+   in lib.warnIf (value != null) warning (if (value != null) then value else newValue);
+```
+
+Starting with following package:
+```
+{ lib
+, stdenv
+, nonCompliantFoo ? true
+}:
+
+# uses nonCompliantFoo
+stdenv.mkDerivation { ... }
+```
+
+the first step of migration is to replace it with the following:
+```
+{ lib
+, stdenv
+, nonCompliantFoo ? null
+, enable_foo ? lib.renamed {
+    oldName = "nonCompliantFoo";
+    newName = "enable_foo";
+    sunset = "25.11";
+    value = nonCompliantFoo;
+  } true
+}:
+
+# uses enable_foo
+stdenv.mkDerivation { ... }
+```
+
+where `sunset` parameter is set two full releases after time of this change.
+So, if this change is made in Jan 2024, `sunset` MUST be set to `25.11`. After
+the release `25.05` all mentions of `nonCompliantFoo` SHOULD be removed, and
+package will look like following:
+
+```
+{ lib
+, stdenv
+, enable_foo ? true
+}:
+
+# uses enable_foo
+stdenv.mkDerivation { ... }
+```
+
+## Feature parameter naming guidelines
```

Comment:
There should be some example with a multi-word package being an optional dependency.

Reply:
Added one.